microcosm_postgres/encryption/v2/column.py

```diff
@@ -71,19 +124,32 @@ def _prop(self):
             return encoder.decode(encryptor.decrypt(encrypted))
 
         def _prop_setter(self, value) -> None:
-            encrypted = encryptor.encrypt(encoder.encode(value))
+            encoded = encoder.encode(value)
+            encrypted = encryptor.encrypt(encoded)
             if encrypted is None:
                 setattr(self, unencrypted_field, value)
                 setattr(self, encrypted_field, None)
+                if hasattr(self, beacon_field):
+                    setattr(self, beacon_field, None)
                 return
 
             setattr(self, encrypted_field, encrypted)
             setattr(self, unencrypted_field, None)
+            if hasattr(self, beacon_field):
+                setattr(self, beacon_field, encryptor.beacon(encoded))
+
+        def _prop_comparator(cls) -> BeaconComparator | None:
+            if beacon := getattr(cls, beacon_field, None):
+                return BeaconComparator(
+                    val=getattr(cls, unencrypted_field),
+                    beacon_val=beacon,
+                    encoder_fn=encoder.encode,
+                    beacon_fn=encryptor.beacon,
+                )
 
-        def _prop_expression(cls):
```

Comment:
Should we keep the expression for when we don't use beacon?

Reply:
We only know whether to use beacon lower down in the call stack e.g in the comparator on an object instance?

I've put more logic in the comparator to deal with using / not using it? I would be simpilar to just to use prop expression though if you know a way?